test/check-pipeline-status.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+PRJ_ROOT="$(cd `dirname "${BASH_SOURCE}"`/..; pwd)"
+
+ENV_CODE=${1:-${ENV_CODE}}
+PIPELINE_RUN_ID=${2:-${PIPELINE_RUN_ID}}
+SYNAPSE_WORKSPACE_NAME=${3:-${SYNAPSE_WORKSPACE_NAME}}
+SLEEP_TIME_BETWEEN_CHECK_STATUS=${4:-${SLEEP_TIME_BETWEEN_CHECK_STATUS}}
+
+if [[ -z "$ENV_CODE" ]]
+  then
+    echo "Environment Code value not supplied"
+    exit 1
+fi
+
+if [[ -z "$SYNAPSE_WORKSPACE_NAME" ]]; then
+    SYNAPSE_WORKSPACE_NAME="${ENV_CODE}-pipeline-syn-ws"
+fi
+if [[ -z "$SLEEP_TIME_BETWEEN_CHECK_STATUS" ]]; then
+    SLEEP_TIME_BETWEEN_CHECK_STATUS=60
+fi
+
+set -ex
+
+PIPELINE_RUN_STATUS=$(az synapse pipeline-run show --workspace-name ${SYNAPSE_WORKSPACE_NAME} --run-id ${PIPELINE_RUN_ID} --query status --output tsv)
+while [[ "$PIPELINE_RUN_STATUS" != "Succeeded" ]] && [[ "$PIPELINE_RUN_STATUS" != "Failed" ]]
```

Comment:
@mandarinamdar The pipeline run time is a key metrics that we can capture and log for every run. For now, can we log this metrics inside the pipeline? Long-term may be have a App Insights setup to capture the run times for the pipelines.

Thoughts?

Reply:
Yes, agreed. To start with if we can just log it as part of CI is still good info to capture. Later plan to incorporate that with app insights / az monitor

Reply:
Not sure, if we want to publish the time at the end of the script how much time it took.

Reply:
Will add this as a backlog item for future sprint.